### PR TITLE
generate csv using go code, added disconnected installation support

### PR DESCRIPTION
**Pull request by redhatHameed**
Signed-off-by: Abdul Hameed <ahameed@redhat.com>

### CODE REVIEW

**Comment by redhatHameed** [comment]
What this PR does:
This PR provides initial offline installation support of apicurio-operators:

  Generate the CSV file now using go code
   Added required configuration for offline support in CSV          
   Fetch the digest from the registry and replace the tags, the code only support to fetch the digest from redhat.io registry

File: apicurito/tools/components/components.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+	"strings"
+	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var Verbs = []string{
```
Comment (lgarciaaco) [inline apicurito/tools/components/components.go]: I think this doesn't need to be accessed outside the package

**Review by lgarciaaco — APPROVED** [review]
Hi Hameed, nice work

File: apicurito/tools/components/components.go (hunk 1 above)
Comment (redhatHameed) [reply apicurito/tools/components/components.go]: Fixed

**Comment by lgarciaaco** [comment]
Maybe it sense to include this as a command. Currently we only have `run`, you can see it here: https://github.com/Apicurio/apicurio-operators/blob/master/apicurito/pkg/cmd/public.go

**Comment by redhatHameed** [comment]
> Maybe it sense to include this as a command. Currently we only have `run`, you can see it here: https://github.com/Apicurio/apicurio-operators/blob/master/apicurito/pkg/cmd/public.go

@lgarciaaco I already added the command in make file to generate csv like, 
```bash
make csv

# OR
# w/ sha lookup/replacement against registry.redhat.io
DIGESTS=true REDHATIO_TOKEN="<username>:<password>"  make csv
```
So still we need to add CSV command?  and also how this `run` command work, I mean how you call this.

**Comment by lgarciaaco** [comment]
> > Maybe it sense to include this as a command. Currently we only have `run`, you can see it here: https://github.com/Apicurio/apicurio-operators/blob/master/apicurito/pkg/cmd/public.go
> 
> @lgarciaaco I already added the command in make file to generate csv like,
> 
> ```shell
> make csv
> 
> # OR
> # w/ sha lookup/replacement against registry.redhat.io
> DIGESTS=true REDHATIO_TOKEN="<username>:<password>"  make csv
> ```
> 
> So still we need to add CSV command? and also how this `run` command work, I mean how you call this.

@redhatHameed right. For other middle-ware operators we use the operator binary as a command line. For apicurito there is only one command(run) since this a very simple operator. This is the default command that runs the manager and the one it is called when apicurito is deployed, see https://github.com/Apicurio/apicurio-operators/blob/3b6f03705745266665f2039023d40ee53f4adc44/apicurito/build/Dockerfile#L14

It would be nice if we can generate the bundle files using the same binary instead of `make`, something like: `./apicurito-operator csv`.

I think we could merge this PR since it is open for long now, and add this change later, WDYT?

**Pull request merged**